bokeh/models/tests/test_sources.py

```diff
@@ -289,15 +323,17 @@ def stream_wrapper(*args, **kwargs):
         self._assert_equal_dicts_of_arrays(notify_owners_stuff['args'][0],
                                            dict(a=np.array([10]),
                                                 b=np.array([20]),
-                                                c=np.array([30])))
+                                                c=np.array([30]),
+                                                index=np.array([0])))
 
         self._assert_equal_dicts_of_arrays(dict(ds.data),
                                            dict(a=np.array([10, 11]),
                                                 b=np.array([20, 21]),
-                                                c=np.array([30, 31])))
+                                                c=np.array([30, 31]),
+                                                index=np.array([0, 0])))
```

Comment:
Here is an example of an index column in a CDS getting duplicates. This is not new, and I don't think there is anything that can be done about it. If users want the CDS index column to really be an index with unique valuea at all times, that is up to them to enforce. Though perhaps I could add documentation support to call out the possibility explicitly.